components/director/internal/repo/exist.go

```diff
@@ -42,11 +44,18 @@ func NewExistQuerierGlobal(resourceType resource.Type, tableName string) ExistQu
 }
 
 // Exists missing godoc
-func (g *universalExistQuerier) Exists(ctx context.Context, tenant string, conditions Conditions) (bool, error) {
-	if tenant == "" {
+func (g *universalExistQuerier) Exists(ctx context.Context, tenantID string, conditions Conditions) (bool, error) {
+	if tenantID == "" {
 		return false, apperrors.NewTenantRequiredError()
 	}
-	conditions = append(Conditions{NewTenantIsolationCondition(*g.tenantColumn, tenant)}, conditions...)
+
+	switch tenant.LoadIsolationTypeFromContext(ctx) {
+	case tenant.RecursiveIsolationType:
```

Comment:
instead of switching in every universal storage method, you could pass the context to NewTenantIsolationCondition and let it calculate the isolation query (encapsulate the switch in it, in a single place)